### PR TITLE
Overview api calls setup

### PR DESCRIPTION
**Pull request by dwightdoran**
Apologies for this but there is a lot of changes here....

-I added all of the routes from the client --> server --> API
-Refactored my API functions (added callback as parameter)
-Added all of the routes I am going to need to the server besides the cart post route i'll need (not sure of how i want to implement that yet)
-The purpose of this branch was just to set up the routes from the client to the API and back // make sure each component is getting the data it needs and nothing more.

Again apologies for the large PR (got in the zone and didn't realize how much I was writing)

Also I had to cancel my last PR because I left my key in there. I'm creating a new one now but I'd like to get that figured out soon.

### CODE REVIEW

**Review by RobStafford — APPROVED** [review]
Hi Dwight--I see two potential issues, but neither I'd request a change on: 1) if we all store our api token in the same variable (Dominic and I had picked GITHUB_API_TOKEN, but idc) then the one import statement will work for all four of us, without a need for individual token imports, and 2) you've hard coded an item id in to the props you're passing down from src/index.jsx to Overview, I'd suggest you put it as the default in state, instead, as the component will still function correctly when state changes, and you'll get the behavior you want for testing--that's what I did in my incoming repo).

File: utils/apiCalls.js [hunk 1]
```diff
@@ -63,4 +129,4 @@ module.exports.getProductDataByItem = getProductDataByItem;
 // module.exports = {
 //   getReviewsByItem,
 //   getReviewsMetaByItem
-// }
+// }
```
Comment (DominicMonares) [inline utils/apiCalls.js:147]: newline needed

File: client/src/index.jsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ class App extends React.Component {
       <div>
         <h1>Atelier</h1>
         <div className="overview">
-          <Overview itemid={this.state.item_id} />
+          <Overview itemid={59553} />
```
Comment (DominicMonares) [inline client/src/index.jsx]: I think Rob's suggestion to put a hardcoded id in the App state is a good one, that way we can keep everything dynamic. I'm going to make that same change to my QA-framework

File: client/src/components/Overview/components/StyleSelector.jsx [hunk 3]
```diff
@@ -16,4 +16,4 @@ class StyleSelector extends React.Component {
   }
 }
 
-export default StyleSelector;
+export default StyleSelector;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/StyleSelector.jsx]: newline needed

File: client/src/components/Overview/components/ProductInfo.jsx [hunk 4]
```diff
@@ -1,19 +1,32 @@
...
+
   render() {
     return (
       <div className="product-info">Product Info</div>
     )
   }
 }
 
-export default ProductInfo;
+export default ProductInfo;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/ProductInfo.jsx]: newline needed

File: client/src/components/Overview/components/AddToCart.jsx [hunk 5]
```diff
@@ -5,15 +5,19 @@ class AddToCart extends React.Component {
...
+
   render() {
     return (
       <div className="add-to-cart-info">Add To Cart Area</div>
     )
   }
 }
 
-export default AddToCart;
+export default AddToCart;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/AddToCart.jsx]: newline needed

File: client/src/components/Overview/Overview.jsx [hunk 6]
```diff
@@ -1,34 +1,55 @@
...
+          <AddToCart styleData={this.state.styleData} />
+          <ImageGallery styleData={this.state.styleData}/>
         </div>
       </div>
     )
   }
 }
 
-export default Overview;
+export default Overview;
```
Comment (DominicMonares) [inline client/src/components/Overview/Overview.jsx]: newline needed

**Review by DominicMonares — CHANGES_REQUESTED** [review]
Hey Dwight, looks great overall! Only changes I see are the newlines at the end of some docs again (please let me know if this is too nitpicky, I know it doesn't affect the code but github picks up on it) and the hardcoded item id.

I did the same thing for my id but I think Rob's suggestion to put it as a default state is a good one. No need to change the state in index.jsx, I'm putting the default on my QA-framework PR, just the reference to the state.

We should all discuss the API key setup when we meet tomorrow, but for now I'm good with it as is.

**Comment by dwightdoran** [comment]
Thank you for the suggestions gentlemen. I have made the changes and am awaiting your approval good sirs.

**Pull request merged**